services/meeting_service.js

```diff
@@ -60,29 +71,57 @@ const meetingService = {
             checkData = Checker.checkDataWithUUID(uuid,hostsMails,guestMail,duration)
             meeting = await Meeting.create({uuid:uuid, duration: duration})
         }
-        if(checkData!==true) return createResponse(checkData, BAD_REQUEST_CODE)
+        if(checkData!==true) return RestUtils.createResponse(checkData, RestUtils.BAD_REQUEST_CODE)
         const guest = await findOrCreateGuest(guestMail)
         const hosts = await findOrCreateHosts(hostsMails)
         await meeting.setGuest(guest)
         await meeting.setHosts(hosts)
-        return {...createResponse("Meeting added"), uuid: meeting.uuid}
+        return {...RestUtils.createResponse("Meeting added"), uuid: meeting.uuid}
     },
         
 
     updateMeeting: async (uuid, hostsMails, guestMail, duration) => {
         const meeting = await Meeting.findByPk(uuid)
         const checkData = Checker.checkDataWithUUID(uuid,hostsMails,guestMail,duration)
-        if(checkData!==true) return createResponse(checkData, BAD_REQUEST_CODE)
-        if(meeting.startTime!==null) return createResponse("You can't update scheduled meeting",BAD_REQUEST_CODE)
+        if(checkData!==true) return RestUtils.createResponse(checkData, RestUtils.BAD_REQUEST_CODE)
+        if(meeting.startTime!==null) return RestUtils.createResponse("You can't update scheduled meeting", RestUtils.BAD_REQUEST_CODE)
         await updateMeetingHosts(meeting,hostsMails)
         await updateMeetingGuest(meeting,guestMail)
         await updateMeetingDuration(meeting,duration)
-        return createResponse("Meeting updated")
+        return RestUtils.createResponse("Meeting updated")
+    },
+
+    deleteMeeting: async (uuid) => await Meeting.destroy({where: {uuid: uuid}}),
+
+    getTimeSlotsIntersection: async (uuid) => {
+        const meeting = await getMeetingWithHosts(uuid)
+        if (meeting == null) return RestUtils.createResponse("No meeting with such ID!", RestUtils.NOT_FOUND_CODE)
+        let slots = []
+        for (const host of await meeting.getHosts()) {
+            const h = await HostService.getHostWithTimeSlots(host.uuid)
+            slots.push(h.host.TimeSlots)
+        }
+        slots.forEach(s => s.sort((a, b) => a.startDatetime < b.startDatetime))
+        const intersection = TimeSlotsUtils.getIntersection(slots)
+        return {...RestUtils.createResponse("Available timeslots"), timeSlots: intersection}
     },
 
-    deleteMeeting: async (uuid) => await Meeting.destroy({where: {uuid: uuid}})
-    
+    pickTimeSlot: async (uuid, req) => {
+        await Meeting.update({startTime: req.startTime}, {
+            where: {
+                uuid: uuid
+            }
+        });
+        const meeting = await getMeetingWithHosts(uuid)
+        // console.log(meeting)
+        for (const host of meeting.Hosts) {
+            const hostResponse = await HostService.getHostWithTimeSlots(host.uuid)
```

Comment:
nazwałbym tę zmienną `hostsWithTimeSlots` - hostResponse kojarzy mi się trochę, jakbys jakiegoś zewnętrznego serwisu pytał.

Reply:
z tym że to jest cały response, opakowany z statusem i msg, z którego trzeba wyciągnąć hosta, a z tego timeSlots, wydaje mi się, że łańcuszek `hostWithTimeSlots.host.timeSlots` nie prezentuje się najlepiej